src/librustc_trans/context.rs

```diff
@@ -343,6 +343,18 @@ impl<'b, 'tcx> SharedCrateContext<'b, 'tcx> {
     pub fn use_dll_storage_attrs(&self) -> bool {
         self.use_dll_storage_attrs
     }
+
+    // LLVM has a separate type for 64-bit SIMD vectors on X86 called `x86_mmx`
+    // which is needed for some SIMD operations. As a bit of a hack (all SIMD
+    // definitions are super unstable anyway) we recognize any one-element SIMD
+    // vector as "this should be an x86_mmx" type. In general there shouldn't be
+    // a need for other one-element SIMD vectors, so it's assumed this won't
+    // clash with much else.
+    pub fn use_x86_mmx_for_vector(&self, count: usize, size: Size) -> bool {
+        let x86 = self.tcx.sess.target.target.arch == "x86" ||
+                  self.tcx.sess.target.target.arch == "x86_64";
+        count == 1 && size.bits() == 64 && x86
```

Comment:
Can you inline this, including the comment, into `type_of`?

Reply:
Certainly, done now